src/main/java/io/outfoxx/swiftpoet/CodeWriter.kt

```diff
@@ -443,7 +447,7 @@ internal class CodeWriter(
   /**
    * Returns the non-colliding importable types and module names for all referenced types.
    */
-  private fun generateImports(): Pair<Map<String, DeclaredTypeName>, Set<String>> {
+  private fun generateImports(): Pair<Map<String, List<DeclaredTypeName>>, Set<String>> {
```

Comment:
Would it be better to do the collision check inside and still only export the actually imported types? Rather than a list with possible collisions?

Not actually sure if that idea works.

Reply:
hmmm I will give it a shot! I see what you are suggesting.

Reply:
@kdubb I attempted it and tried to use day with `referencedTypeNames` but that broke a bunch of the tests. 

I think this change gives me the most confidence given all tests pass and the ones that changed made sense ot me.